### PR TITLE
Unconditionally format generated code but in CI

### PR DESCRIPTION
**Pull request by tristan0x**
Rely on utility `cmake/hpc-coding-conventions/bin/format`

### CODE REVIEW

File: src/language/code_generator.py [hunk 1]
```diff
@@ -25,14 +24,20 @@
 
 LOGGING_FORMAT = "%(levelname)s:%(name)s: %(message)s"
 LOGGER = logging.getLogger("NMODLCodeGen")
+CLANG_FORMAT_EXECUTABLE = (
```
Comment (pramodk) [inline src/language/code_generator.py:27]: a pure nitpick: it's not  really a`clang-format` executable?

File: src/language/code_generator.py [hunk 2]
```diff
@@ -98,6 +101,14 @@ def __new__(cls, base_dir, clang_format=None):
...
         return self
 
+    @property
+    def within_ci(self):
+        """Return True if code is running in a continuous-integration environment, else otherwise"""
+        for varname in ["CI", "CI_PIPELINE_ID", "AZURE_ENV_NAME"]:
+            if varname in os.environ:
+                return True
+        return False
+
```
Comment (pramodk) [inline src/language/code_generator.py:111]: wondering why we need it i.e. what happens if the files are formatted in CI as well? 🤔

File: src/language/code_generator.py [hunk 3]
```diff
@@ -287,9 +298,9 @@ def format_output(self, file):
             language: c++ or cmake
 
         """
-        if self.language == "c++" and self.app.clang_format:
-            LOGGER.debug("Formatting C++ file %s", str(file))
-            subprocess.check_call(self.app.clang_format + ["-i", str(file)])
+        if self.language == "c++" and not self.app.within_ci:
```
Comment (pramodk) [inline src/language/code_generator.py:301]: A clarification question: if user doesn't have "compatible" clang format installed on his/her machine, what happens in this case? .i.e. if there is no clang-format installed or of a different version than we support, user will still be able to build the nmodl without any error?

**Review by pramodk — COMMENTED** [review]
Just few clarification questions otherwise LGTM

File: src/language/code_generator.py (hunk 1 above)
Comment (ohm314) [reply src/language/code_generator.py:27]: `CLANG_FORMAT_COMMAND` then :)

File: src/language/code_generator.py (hunk 3 above)
Comment (ohm314) [reply src/language/code_generator.py:301]: I think this is taken care of from within the coding convention tool that wraps clang-format?

File: src/language/code_generator.py (hunk 2 above)
Comment (ohm314) [reply src/language/code_generator.py:111]: I tried to understand this but couldn't quite figure it out from looking at our current CMake files and CI scripts...

**Pull request closed without merging**